### PR TITLE
Xcavate W3F Grant Application Phase 2

### PR DESCRIPTION
**Pull request by xcavate2022**
#### Project Abstract

> Real world real estate has traditionally been only available to a small number of people due to the incredibly complex nature of the asset. It is currently not possible to invest directly from a real estate developer without "trusted" third party intermediaries like banks, solicitors, estate agents and investment funds.
> Our goal is to create a community-controlled protocol, which empowers global communities to directly fund and invest in real estate developments that span across; residential, commercial and community non-for-profit projects. By leveraging sovereign digital identity and real estate asset specific verifiable credentials, this will create a less trust more transparency environment, removing many barriers to entry for the average person.
>
> This is a follow-up to a previous grant, https://github.com/w3f/Grants-Program/pull/1649

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [x] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [x] Payment details have been provided (bank details via email _or_ Polkadot (USDC & USDT) or BTC address in the application).
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [ ] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@_______:matrix.org` (change the homeserver if you use a different one)

### CODE REVIEW

**Review by meanking — APPROVED** [review]

**Comment by meanking** [comment]
How can I sign the CLA?

**Comment by meanking** [comment]
I have read and hereby sign the Contributor License Agreement.

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks a lot for the application. I have a few initial comments/questions: Could you add to the milestone table only things that you need to develop that don't already exist and remove all other parts? For example, we already have multiple NFT pallets in the ecosystem. If you plan to develop something unique (e.g., a new kind of NFT pallet), could you specify how it's different from the existing solutions? For front-end-specific deliveries, we usually ask teams to provide mock-ups/initial designs. Also, there have been multiple issues with your previous delivery. How do you plan to improve the quality of the delivery?

**Comment by xcavate2022** [comment]
Hi David,
Many thanks for your message. I firstly want to apologise for the historical issues relating to our initial application and milestone deliverables. These issues must have been frustrating for the committee and made things more complex and time consuming than they needed to be. When I think back to our first application, I’m sorry to say I relied much too much on the advice of consultants (who in hindsight had their own priorities). Once we took full control of the project ourselves and built our own team, we then had some teething issues around team dynamics and skill sets so a few team member changes were necessary. Now we have a solid team and the skills complement each other really well. I must admit that when it came to the deliverables for milestone 1 we didn’t have enough structure with our documentation and repo management, which were exacerbated with node template update issues. I’m pleased to say that we have a much better discipline and our gitbook documentation structure just needs additional pages and sections for milestone 2. I really do believe our project is in a great position now and the team is going from strength to strength. I do understand your comments regarding the deliverables and after speaking to the team we will revise them to be more specific about what we will deliver (like the NFT pallet should actually be called "Xcavate NFT marketplace pallet” as its more about how the NFTs are presented and the interaction with the various user roles to protect the investor). We will also split the "Xcavate Substrate Node" in to specific pallets, which are now in their second version with added functionality. In addition, we have considerable Figma design .jpg’s I can include in the application and reference them in the frontend deliverable, so it’s easier to measure when that has been achieved.
Lastly, I just wanted to add that I am totally committed to getting this project to market as I feel it could tackle a huge problem in society relating to real estate ownership and utilisation. I am keen to partner with as many ecosystem projects as possible brings new users and liquidity in to the Dotsama world. You only need to see what has happened to Lido Finance to understand without solid projects with sustainable business models they die.
Sorry to ramble on. I hope I've covered your points to your satisfaction? if so, I’ll amend the application and await your response.
All the best
Richard

**Review by takahser — CHANGES_REQUESTED** [review]
@xcavate2022 thanks for your answer. I'd like to highlight that @Noc2's concerns regarding more detailed specs in the deliverables as well as the removal of existing functionality haven't been addressed yet.

In addition to that, I have a few comments as well:

1.  Could you comment on the monthly FTE rate that doubled in comparison to your previous grant?
2. How do you bridge the _land title deed NFT_ to its real-world counterpart?
  image
3. On the 2ndary market, can anybody purchase NFTs? Or are there additional requirements, such as KYC/AML checks? If yes, how does this mechanism work?
  image
4. How does the property valuation process work? Is its outcome better than the market outcome, and if yes, why?
  image
5. Are you sure that your solution is cheaper than what banks provide? After all, banks use fractional reserve banking to [multiply](https://www.investopedia.com/ask/answers/062615/what-difference-between-deposit-multiplier-and-money-multiplier.asp) their liquidity, which makes it really cheap and easy for them to create new units of currency. Quote from your proposal:
    > access to finance through traditional banks and other financial institutions is complex, time consuming and expensive

**Comment by xcavate2022** [comment]
Hi Seraya,

Many thanks for your message.

Yes I was just waiting for a response then I was going to update the application with the following changes to the deliverables:

1.	Xcavate dApp Frontend	User interface to connect wallet and interact with dApp - See example Figma Designs
2.	Xcavate dApp Backend	Connects dApp to substrate node
3.	Community Loan Pallet v2.0	Changing/Adding functionality regarding the loan process, higher dependency on voting and implementing the previous smart contract functionality into to the loan pallet for greater security and performance.
4.	NFT Marketplace Pallet	Has restricted functionality to list NFTs on marketplace only once ownership and price checks have been successfully completed.
5.	Governance Functionality v2.0	Implementing greater democracy into the loan pallet so that our loan technical committee can vote on milestones, loan proposals and be able to delete the loan when loan has been repaid.



Regarding your comments - please see my response:
1. Apologies I think both our milestone applications had incorrect FTE… Milestone 1 should have been 1.25 and milestone 2 should be 2.5 (this will be amended as part of the application update).
2. Once ownership of the land has been established and verified (a combination of Oracle API calls & KYC/KYB credentials). There would be a 1st charge applied to the land (which would be removed once the loan has been repaid).
3. All participants of the platform would need to comply with KYC & AML (we are in talks with Deloitte about self sovereign verifiable credentials)
4. The market price for real estate (along with potential rental income) is normally initially set between the real estate developer and an estate agent… our protocol takes the real estate developers estimate and checks against industry data for that type of property in the geographic location (within a certain tolerance) and does not allow listing unless price is fair… giving the investor confidence that the real estate developer is not over charging the potential investor… then market forces / investor choice would indicate whether this is acceptable.
5. Banks are currently offering UK small to medium sized real estate developers 12% to 15% APR, showing they are profit driven not community driven (small to medium sized developments are much better for the community they are built in, yet banks give larger developers preferential rates)… our protocol would be about 9% to 12% APR (7% to 10% of which would go to the community loan staking participants… much more favourable interest than saving in a bank… then the loan arrangement fees would be delivered back to the community loan pallet to help finance more loans / community approved projects) plus be processed in days rather than weeks or months, so the real estate developers can build more projects helping to maintaining market equilibrium. Also the loan set up costs would be considerably cheaper with our solution than a bank or private lending solution.

Hope this helps with your grant application decision.

Best Regards,

Richard

**Comment by takahser** [comment]
@xcavate2022 thanks for the quick reply.

1. sounds good
2. I didn't understand how that mechanism works. Can you elaborate and explain step-by-step?
3. OK
4. Oh, so that means that applies only for the initial price? And later, in the secondary market it will be based on demand and supply, right?
5. 12-15% APR, where did you get these numbers from? The current[ base rate seems to be at 5.25%](https://commonslibrary.parliament.uk/economic-update-have-interest-rates-peaked/#:~:text=The%20Bank%20of%20England%20left,not%20need%20to%20increase%20further.) and the fixed-term mortgage rates are roughly between [5-6.4%](https://www.rightmove.co.uk/news/articles/property-news/current-uk-mortgage-rates/), as per 17 Oct 2023:
  image

**Comment by xcavate2022** [comment]
Hi Seraya,

The lending protocol is not based on mortgages… it’s based on funding construction costs for a real estate development.

An example would be a real estate developer who owns a plot of land with planning permission for 9 apartments. Say they need to borrow £1,000,000 to build the finished properties. Typically the land would be worth around £1,000,000 (which would be used as collateral against the loan).
Then each apartment might cost £333,333 each when completed… so a GDV (gross development value) of £3,000,000

The real estate developer would fill out a loan application, which would be voted on by a loan committee (which are appointed by the community - based on having certain experience / credentials)… later this might be fully automated using an algorithm if approved by the community after significant successful testing.

If the loan is approved then a personalised APR is offered on the loan based on a formula, which risk rates the loan (while bank base rate is used in construction build loans, there are a number of factors including; loan term & the developers experience).

If the loan terms & conditions are accepted then the full loan amount is locked in an account along with the land NFT held as collateral. Based on the build schedule the first tranche of money is delivered to the real estate developers multisig wallet… then once evidence of the first milestone has been supplied & approved by majority votes by the committee then the next tranche would be transferred. This continues until all the properties have been built (interest is calculated on a daily rate and added to the initial loan amount).

Then to repay the loan the real estate developer can mint each property as a collection x100 NFTs, which are then listed on the NFT marketplace. Once all NFTs have been sold an SPV (special purpose vehicle) is created (property title deed is transferred from real estate developer to the SPV) and the property is now owned by the NFT holders (a one for one share with the SPV). Once the loan has been repaid with the proceeds of the sale (loan amount plus interest) then any surplus is transferred to the multisig of the developer as profit.

Each property is then allocated to a pre approved letting agent and the NFT holders receive rental income based on the number of NFTs they hold (less a management fee and taxes).

Then later if the NFT holder wishes to sell the NFT they firstly get offered to the renter of the property, then the other NFT holders of that property… then finally re-listed on the NFT marketplace (at market rate - based on the oracle price feed).

NB. The last two paragraphs of the process are not part of milestone 2… they would be included in milestone 3 if we require further grant support.

Best Regards,

Richard

**Comment by takahser** [comment]
@xcavate2022 

> An example would be a real estate developer who owns a plot of land with planning permission for 9 apartments.

Does that mean you're not planning to give loans for land, but only for construction on land that is already owned by the developer?

> If the loan terms & conditions are accepted then the full loan amount is locked in an account along with the land NFT held as collateral. 

Can you describe how the ownership of the land NFT is coupled with the ownership of the property title deed as registered at the [HM Land Registry](https://www.gov.uk/government/organisations/land-registry) or similar registers?

> Once the loan has been repaid with the proceeds of the sale (loan amount plus interest) then any surplus is transferred to the multisig of the developer as profit.

What if the loan amount was too low and the costs exceed the expectations? Doesn't that happen quite often?

> Then later if the NFT holder wishes to sell the NFT they firstly get offered to the renter of the property, then the other NFT holders of that property… then finally re-listed on the NFT marketplace (at market rate - based on the oracle price feed).

That sounds not very attractive to the NFT holder. Why not ask for any price he wants? The oracle price feed might be incorrect, as real estate prices often vary highly on a number of variables that can not all be taken into account accurately by price models. Examples include the view (maybe it's on the 3rd floor and the oracle increases the price, however, the view is blocked by a higher building), noise (maybe on one side of the building it's more noisy than on the other), black swan events, etc.

Finally, I'd like to hear the benefits of using blockchain over some centralised services - after all it seems like ownership of the flats are determined by the ownership of the corresponding SPV - linking the NFT to the SPV share might be difficult and doesn't come with many benefits, especially if you KYC/AML any participant anyway.

Btw, you didn't address the interest rates question from my [previous comment](https://github.com/w3f/Grants-Program/pull/2054#issuecomment-1776055860):

> 12-15% APR, where did you get these numbers from? The current[ base rate seems to be at 5.25%](https://commonslibrary.parliament.uk/economic-update-have-interest-rates-peaked/#:~:text=The%20Bank%20of%20England%20left,not%20need%20to%20increase%20further.) and the fixed-term mortgage rates are roughly between [5-6.4%](https://www.rightmove.co.uk/news/articles/property-news/current-uk-mortgage-rates/), as per 17 Oct 2023:

File: applications/Xcavate2.md
Comment (takahser) [reply applications/Xcavate2.md:151]: FYI - that's just for formatting

File: applications/Xcavate2.md
Comment (takahser) [inline applications/Xcavate2.md:167]:
```suggestion
| 2. | Xcavate dApp Backend   |  Connects dApp to substrate node |

```

File: applications/Xcavate2.md
Comment (takahser) [inline applications/Xcavate2.md:170]: Isn't that part of the loan pallet? Can you merge this with _3. | Community Loan Pallet v2.0_?

**Comment by xcavate2022** [comment]
> An example would be a real estate developer who owns a plot of land with planning permission for 9 apartments.
> 
> Does that mean you're not planning to give loans for land, but only for construction on land that is already owned by the developer?
> 
Yes
> If the loan terms & conditions are accepted then the full loan amount is locked in an account along with the land NFT held as collateral.
> 
> Can you describe how the ownership of the land NFT is coupled with the ownership of the property title deed as registered at the HM Land Registry <https://www.gov.uk/government/organisations/land-registry> or similar registers?
> 
So once KYC/KYB/AML/CreditScore checks have been completed (so we know who the loan applicant is (as 99% of the time the land will be owned a company / 1% by an individual)… then an Oracle will check land registry (and valuation) to confirm that company is correct (and there is currently no “charge" on the land)… once these things have been established and authorisation given to use land as collateral by the owners (e-sign T&Cs with DIDs), then this meta data will be part of the land loan collateral NFT along with the 1st charge on the land (so if there was a loan default the land can be allocated to another real estate developer to finish the development and repay the loan or simply sold to repay the loan).
> Once the loan has been repaid with the proceeds of the sale (loan amount plus interest) then any surplus is transferred to the multisig of the developer as profit.
> 
> What if the loan amount was too low and the costs exceed the expectations? Doesn't that happen quite often?
> 
If something happens with the project and there is a requirement to borrow additional funds to finish the development, then as long as the collateral land value allowed for that and the loan technical committee voted to accept the risks, an additional tranche of loan funds could be added to the loan… this can happen if building materials / labour costs increase considerably post application but mostly it is priced in to the build schedule. If it was voted down by the committee (because its deemed to be project mismanagement) then loan default measures would be implemented. 

Then later if the NFT holder wishes to sell the NFT they firstly get offered to the renter of the property, then the other NFT holders of that property… then finally re-listed on the NFT marketplace (at market rate - based on the oracle price feed).
> That sounds not very attractive to the NFT holder. Why not ask for any price he wants? The oracle price feed might be incorrect, as real estate prices often vary highly on a number of variables that can not all be taken into account accurately by price models. Examples include the view (maybe it's on the 3rd floor and the oracle increases the price, however, the view is blocked by a higher building), noise (maybe on one side of the building it's more noisy than on the other), black swan events, etc.
> 
Sorry if this is not clear… the selling price is determined by the NFT seller (but checked against real market valuations to protect the buyer). The offer to those parties just makes for the quickest chances of a sale, as those parties have the most interest in that particular property. We would implement a DID communication method so speed of sale was maintained without each party needing to know the identity of the other.
> Finally, I'd like to hear the benefits of using blockchain over some centralised services - after all it seems like ownership of the flats are determined by the ownership of the corresponding SPV - linking the NFT to the SPV share might be difficult and doesn't come with many benefits, especially if you KYC/AML any participant anyway.
> 
Blockchain provides immutable records and less need to trust centralised parties… property fraud causes misery to investors so by recording all transactions on chain, significantly reduces the chance of fraud occurring.

Introducing NFTs offers much greater scope to redefine property ownership and utility in the future… imagine a property club dApp where the NFTs gave you rights to live or work for a specified period of time in any property that was available (within the club) throughout the world. Imagine Hotels being built and NFT holders having the choice to either take holidays or rental income. Also imagine that a village in Africa needs a school built… with our protocol people could donate to the project regardless where they live having total transparency of build and allocation to a non for profit to run. Maybe a community decides they need a homeless shelter… the protocol could provide the transparent and decentralised mechanism to initiate these at scale anywhere in the world. 

The protocol we are building is way of bridging real world real estate and the digital one to enhance peoples lives.
> Btw, you didn't address the interest rates question from my previous comment <https://github.com/w3f/Grants-Program/pull/2054#issuecomment-1776055860>:
> 
> 12-15% APR, where did you get these numbers from? The current base rate seems to be at 5.25% <https://commonslibrary.parliament.uk/economic-update-have-interest-rates-peaked/#:~:text=The%20Bank%20of%20England%20left,not%20need%20to%20increase%20further.> and the fixed-term mortgage rates are roughly between 5-6.4% <https://www.rightmove.co.uk/news/articles/property-news/current-uk-mortgage-rates/>, as per 17 Oct 2023:
> 
You will not get high street banks advertising real estate development loans like retail mortgage loans. However, if you look at this website it gives you a guide. The thing with a real estate development loan is that the headline APR is only one aspect of the loan… you also have loan arrangement fees, broker fees, exit fees and professional fees… all these elements can add significant time to the application being approved / rejected as well as adding cost, which can mean the developer misses out on the development opportunity.

https://abcfinance.co.uk/development-finance/rates-and-fees/#:~:text=Loans%20tend%20to%20start%20at,9%25%20per%20annum%20are%20common.

By making the process transparent, simple and easy to follow and by having a route to market as a way to repay the loan, it is very attractive option to a real estate developer. While also providing a constant flow of new well vetted properties to the NFT marketplace investors.

**Comment by xcavate2022** [comment]
Sorry if I’ve got this wrong but we have a team of six working on this full time so that is the total cost associated with that amount of work in the period of time it would take… when you questioned it I looked at other similar projects and they had the same amount of W3F grant with a lower FTE so I adjusted ours accordingly.

**Comment by xcavate2022** [comment]
I think I understand what you mean now… in the previous milestone 1 we use ink! smart contract, so this was much easier and quicker to create and implement. However, in milestone 2 we have created custom pallets, which take much more time to construct, test and validate.

**Review by takahser — CHANGES_REQUESTED** [review]
See inline comments.

File: applications/Xcavate2.md
Comment (takahser) [reply applications/Xcavate2.md:168]: replaced by https://github.com/w3f/Grants-Program/pull/2054#discussion_r1370213110

File: applications/Xcavate2.md
Comment (takahser) [reply applications/Xcavate2.md]: replaced by https://github.com/w3f/Grants-Program/pull/2054#discussion_r1370209195

File: applications/Xcavate2.md
Comment (takahser) [reply applications/Xcavate2.md]: replaced by https://github.com/w3f/Grants-Program/pull/2054#discussion_r1370222649

File: applications/Xcavate2.md
Comment (xcavate2022) [reply applications/Xcavate2.md:140]: Sorry - I don't understand this suggestion as its definitely milestone 2... why are you suggesting its milestone 1?

Comment (takahser) [reply applications/Xcavate2.md:140]: Can you point me to milestone 1 then?

Comment (takahser) [reply applications/Xcavate2.md:140]: Previous proposal see [comment](https://github.com/w3f/Grants-Program/pull/2054?notification_referrer_id=NT_kwDOAFJNKLI4MDgzOTcwMzQ3OjUzOTM3MDQ&notifications_query=is%3Aunread#issuecomment-1777539556). Usually teams would just add "phase 2" or something similar to their follow-up proposal and then start the milestones at "1" again, since it's a new grant.

File: applications/Xcavate2.md
Comment (takahser) [inline applications/Xcavate2.md:1]:
```suggestion
# Xcavate - Phase 2

```

**Review by takahser — CHANGES_REQUESTED** [review]
I think the easiest way forward is to:

- add "Phase 2" to the title
- change "Milestone 2" to "Milestone 1"

These changes are necessary to avoid confusion. See also my [previous comment](https://github.com/w3f/Grants-Program/pull/2054/files#r1370914871).

**Review by takahser — CHANGES_REQUESTED** [review]
@xcavate2022 I decided to not add my approval to this proposal. These are the reasons:

- the scope seems to be very limited - when compared with your [previous grant](https://github.com/w3f/Grants-Program/blob/master/applications/Xcavate.md), it seems like the main features are
    1. a frontend,
    2. a rewrite of the existing ink! contracts to a pallet and
    3. adding a pallet to allow for a 2ndary market.
- now, while i) is very specific to your proposal and hence the potential reusability by the community limited, I think ii) is nothing very significant and for iii) similar solutions already exist (e.g. _Marketplaces_ in the [tech stack](https://wiki.polkadot.network/docs/build-open-source#about) or the [NFT pallet](https://github.com/paritytech/polkadot-sdk/tree/master/substrate/frame/nfts) that has extrinsics for trading). Overall, I think $30k is a bold price for that.
- I also am not convinced on the mechanism of securing the ownership of the real-world asset through an NFT.
- Furthermore, I'm not sure if blockchain is necessarily the best solution for your business case. That concern is partially due to the previous point - if there was a truly decentralised and robust way to couple the NFT to the real-world asset, I might change my opinion on that.
- Finally, I have concerns regarding your performance in the previous delivery.

That being said, I retract my points I made earlier regarding the interest rates - I understood that mortgage rates should not be mistaken with short-term bridge loans that developers typically seek for and that the rates vary highly. I still don't understand the mechanism that would give your solution a competitive advantage though, because I'd assume the developer can pledge their land as a collateral in a traditional bank-backed loan - so there'd be no difference to your solution.

**Comment by xcavate2022** [comment]
Hi Seraya,
Many thanks for your message. Obviously it’s hugely disappointing that our application hasn’t convinced you of the need for our solution. Its funny because we’ve only recently met up with one of the team at Parity (Tom Humber) who has a project called Archisinal, which is an NFT marketplace for Architects drawings… they were looking to launch on Astar but loved our project and understood its need and they think it would be a good fit with their project, so would happily launch their smart contract on the Xcavate dApp chain.
The challenge (especially in this climate) for projects like ours is to keep our team supported and rewarded so that was the key driver behind the amount of the grant requested. While we have a number of things that we are pursuing this is a pivotal moment. We are hoping to be accepted on the Substrate Builders Program and get Seed investment so they are still within our sights. In the meantime if we revised our application to a more modest amount, would it be something you would reconsider?
Best Regards,
Richard

**Comment by takahser** [comment]
@xcavate2022

> so would happily launch their smart contract on the Xcavate dApp chain.

That's great news for your project!
Obviously, I can't speak for them, but personally, I feel like you haven't made a convincing point yet on why your NFT marketplace should be superior to existing solutions. Hence, my decision.

Btw, my single vote doesn't necessarily mean that the project will be rejected - other committee members can have a different opinion and potentially decide to add their approval.

**Comment by xcavate2022** [comment]
Thank you… Yes I totally understand. Well I’ve just been working with one of my team on the Oracles element… we are using the Acurast Pallet so that Price, Title Deed & Planning Approval are remotely checked by Oracle prior to NFTs being allowed to list (I’m not aware of any other RWA NFT marketplace having this functionality so you have to “Trust” the listing entity)… The reason this is necessary with our solution is that the buyer is buying direct from the builder of the property (not a fund or an agent which is the case with other solutions). This direct relationship makes for a much more efficient system (much quicker from build to market), which means the real estate developers can build more properties and so there is more choice for buyers and property prices are more stable (greater market equilibrium).

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks a lot for the application and all the effort you put into this. However, the grants committee decided not to support your application. Besides the reason @takahser previously listed, we also think the grants program is no longer the best funding mechanism for this project at this stage. Please don't hesitate to let me know if you have any questions, and we wish you all the best for the project.

**Comment by xcavate2022** [comment]
Ok - many thanks for taking the time to review our application. All the best Richard

**Pull request closed without merging**
